src/helm/benchmark/scenarios/cleva_scenario.py

```diff
@@ -0,0 +1,1491 @@
+        text = transformed_data["input"]
+        if "choices" in transformed_data:
+            # This is a multiple-choice task
+            references: List[Reference] = [
+                Reference(Output(text=text), tags=[CORRECT_TAG] if idx in transformed_data["label"] else [])
+                for idx, text in enumerate(transformed_data["choices"])
+            ]
+        else:
+            # This is a generation task
+            correct_answer = transformed_data["label"]
+            references = [Reference(Output(text=answer), tags=[CORRECT_TAG]) for answer in correct_answer]
+
+        instance = Instance(
+            input=Input(text=text),
+            references=references,
+            split=split,
+        )
+        return instance
+
+    def transform_code(self, data: Dict[str, Any], templates: Dict[str, Any], split: str) -> CodeInstance:
+        """
+        Similar to transform method above, transform_code converts a data point in code synthesis scenario in CLEVA
+        to a HELM CodeInstance according to a given CLEVA prompt template.
+        """
+
+        data["prompt"] = templates["input"].format(**data)
+        instance = CodeInstance(
+            input=Input(text=data["prompt"]),
+            references=[
+                CodeReference(
+                    output=Output(text=data["canonical_solution"]),
+                    test_cases=data,
+                    tags=[CORRECT_TAG],
+                )
+            ],
+            split=split,
+        )
+        return instance
+
+    def _apply_all(self, data: dict, templates: dict) -> dict:
+        """
+        This function applies the CLEVA prompt template to a data point.
+
+        Note that this is an in-place operation.
+
+        The logic is as follows:
+        1. It first maps every entry according to a set of predefined mappings in "verbalizer".
+        2. It then stringifies all fields in the given data point, including processing structured data.
+        3. It finally constructs the input string and reference strings.
+        """
+        # If we define a verbalizer, we map all fields before further processing
+        if templates.get("verbalizer", None) is not None:
+            self._mapping_all(data, templates["verbalizer"])
+
+        if "choices" in data.keys():
+            # We take the corresonding choices and apply the `label` template
+            # Note: we do not allow `label` template to access other fields in multi-choice tasks
+            choices = [self._apply(c, templates.get("label", None), label=c) for c in data["choices"]]
+        # We first convert all fields except `input` to strings
+        transformed_data = copy.deepcopy(data)
+        for k, template in templates.items():
+            if k not in ["input", "verbalizer", "meta", "instruction", "label", "answer_context"]:
+                assert k in data, f"Template key `{k}` is not valid!"
+                transformed_data[k] = self._apply(data[k], template, **data)
```

Comment:
If I'm not wrong, you should `continue` if `k` is "input" or "verbalizer" - "input" is only used in the final merge and operates on `transformed_data` rather than `data` (or can `input` recursively include itself?). And "verbalizer" was already used earlier.

Reply:
I am sorry that I don't think I fully understand your valuable suggestion. 

Below, I attach a template example. With the code you commented on here, we will first convert/organize the "choices" field into a string. Then, we will use all the information from the template and from the raw data to complete the "input" field.
```
{
    "choices": {
        "item_separator": "\n",
        "item_template": "{idx}. {item}",
        "item_index": "upper"
    },
    "instruction": "请根据文段内容补全下划线处的成语。",
    "input": "{content}\n",
    "answer_context": "答：（请用选项序号回答）",
    "meta": {
        "language": "zh-Hans",
        "mul_as_gen": true,
        "postprocess": [
            "common",
            "upper"
        ]
    }
}
```

Reply:
Sorry, I misread the code. I think what you did originally is correct.

I would suggest putting an example of "template" in the comments somewhere, so that it is easier to understand.

Reply:
Thanks! We have added examples for both `_apply` and `_apply_all` methods to better illustrate the underlying workflow.